### PR TITLE
Linters for uniform code

### PR DESCRIPTION
**Pull request by galyfray**
Added tools to lint the code to ensure both quality and style consistency across all of us.
 - bandit : this tool checks for security issue.
 - pylint & flake8 helps ensure that the code style  respect the PEP8 and will enlighten small mistakes that could escape us.
 - pydocstyle will check that the documentation is present and is conform to the PEP257. It doesn't check for documentation quality.
 All of these tools have their own output (sadly) : 
- bandit.html
- pylint.txt
- flake8.txt
- pydocstyle.txt

A make command has been created to automatically lint : `make lint`

As a side note any python IDE should provide either in a native form or as a plugin an integrated PEP8 checker making most of the heavy lifting.

I would have liked to have the time to condense all of these outputs in a nice HTML page but I have decided to put this out of scope of this PR. 

Bandit, flake8 and pydocstyle's configurations are finalized and only require validation. However the pylint's configuration require more in depth discussion  as this tool offer functionality that goes further than the usual python conventions.

To be able to get the whole picture I added codacy's tracker which include all (and a bit more) of the tools I listed and should (hopefully) provide information directly in GitHub making PR reviewing easier. It also include some other nice features already used in the CI branch.

Be sure to use `make update` if you already have initialized your development environment.

### CODE REVIEW

**Comment by Nexowo** [comment]
Menfou palu

File: Makefile [hunk 1]
```diff
@@ -66,14 +66,24 @@ reset:
...
+lint:
+	$(CONDA_ACTIVATE)
+	$(PYTHON) -m bandit -r -f html -o bandit.html .
+	$(PYTHON) -m pylint --exit-zero cvrp test >pylint.txt
+	$(PYTHON) -m flake8 . --output-file=flake8.txt
+	$(PYTHON) -m pydocstyle . -e -s > pydocstyle.txt || exit 0
+	
+
+
 help:
```
Comment (m-aspro) [inline Makefile:78]: It doesn't work any more on Windows.
Please, fix the command "make help".
Thanks

**Review by m-aspro — CHANGES_REQUESTED** [review]
Good job!

**Comment by Nexowo** [comment]
English mistakes and typing errors on make help
depencies -> dependencies
commads -> commands
compleate -> complete

**Comment by galyfray** [comment]
Due to the discovery of a major bug I have decided to make this PR a draft until everything works fine.

**Comment by KwassiSenam** [comment]
that's good !!

**Review by KwassiSenam — COMMENTED** [review]
I've seen all changes.

**Comment by KwassiSenam** [comment]
Sorry! I'll let Cyril close this PR if everything is ok.

**Review by Nexowo — APPROVED** [review]

**Review by m-aspro — APPROVED** [review]
Tested 2 days ago on my Windows environment and approved.
Good job!

**Review by Axelvel — APPROVED** [review]

**Pull request merged**